packages/opentelemetry-exporter-jaeger/src/jaeger.ts

```diff
@@ -36,9 +37,22 @@ export class JaegerExporter implements SpanExporter {
     this._onShutdownFlushTimeout =
       typeof config.flushTimeout === 'number' ? config.flushTimeout : 2000;
 
-    config.host = config.host || process.env.JAEGER_AGENT_HOST;
+    // https://github.com/jaegertracing/jaeger-client-node#environment-variables
+    // By default, the client sends traces via UDP to the agent at localhost:6832. Use JAEGER_AGENT_HOST and
+    // JAEGER_AGENT_PORT to send UDP traces to a different host:port. If JAEGER_ENDPOINT is set, the client sends traces
+    // to the endpoint via HTTP, making the JAEGER_AGENT_HOST and JAEGER_AGENT_PORT unused. If JAEGER_ENDPOINT is secured,
+    // HTTP basic authentication can be performed by setting the JAEGER_USER and JAEGER_PASSWORD environment variables.
+    if (config.endpoint) {
+      config.endpoint = config.endpoint || process.env.JAEGER_ENDPOINT;
```

Comment:
the config is being override here with process.env. I think the config entered from a user should treated as something immutable so some copy here should be made before for example
```
const localConfig = Object.assign({}, config);
// and then
localConfig.username = config.username || process.env.JAEGER_USER;
// ...
```

Reply:
I agree we should not mutate the input

Reply:
Fair enough, i will make a pr refactoring it.

Reply:
This should fix it: 2c757923142e778ca2212c8548a50105adb6ca40